src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -443,3 +444,51 @@ setOrResetPltsqlFuncRestoreGUCs(Archive *fout, PQExpBuffer q, const FuncInfo *fi
 			break;
 	}
 }
+
+/*
+ * getCurrentSetting - returns current setting of given guc
+ * Note that, return result is palloc'd which should be freed by caller
+ */
+static char *
+getCurrentSetting(Archive *AH, const char *guc)
+{
+	PGresult *res;
+	PQExpBuffer query;
+	char *setting;
+
+	query = createPQExpBuffer();
+	appendPQExpBuffer(query, "select setting from pg_settings where name = \'%s\';", guc);
+	res = ExecuteSqlQueryForSingleRow(AH, query->data);
+	setting = pg_strdup(PQgetvalue(res, 0, 0));
+
+	destroyPQExpBuffer(query);
+	PQclear(res);
+
+	return setting;
+}
+
+/*
+ * dumpBabelfishSpecificConfig - dump "alter database %S set ... = /'%s/'" for the babelfish specific GUCs for which
+ * the user defined value should be persisted during upgrade e.g., babelfishpg_tsql.server_collation_name and
+ * babelfishpg_tsql.default_locale.
+ */
+void
+dumpBabelfishSpecificConfig(Archive *AH, const char *dbname, PQExpBuffer outbuf)
+{
+	char	*current_server_collation_name = NULL,
+			*current_default_locale = NULL;
+
+	current_server_collation_name = getCurrentSetting(AH, "babelfishpg_tsql.server_collation_name");
+	if (current_server_collation_name)
+	{
+		appendPQExpBuffer(outbuf, "alter database %s set babelfishpg_tsql.restored_server_collation_name = \'%s\';\n", dbname, current_server_collation_name);
```

Comment:
`alter database` won't be effective for the current session. If `restored_server_collation_name` value is only needed while restoring, please consider using just `set babelfishpg_tsql.restored_server_collation_name`.

Reply:
Yes, that is true. But this new function dumpBabelfishSpecificConfig is being called from dumpDatabaseConfig where posgres dumps database level config. This setting is indeed taking effect during upgrade. I have not checked much details why it should work but it seems to be working as expected. 

And, We need to set the value of these GUC at database level because this setting would be useful during minor version upgrade after pg_upgrade.

Reply:
```
pg_restore: connecting to database for restore
pg_restore: creating DATABASE "jdbc_testdb"
pg_restore: connecting to new database "jdbc_testdb"
pg_restore: creating DATABASE PROPERTIES "jdbc_testdb"
pg_restore: connecting to new database "jdbc_testdb"
```